risc0/zkvm/src/session.rs

```diff
@@ -155,3 +160,36 @@ impl SimpleSegmentRef {
         Self { segment }
     }
 }
+
+/// A basic implementation of a [SegmentRef] that saves the segment to a file
+///
+/// The [Segment] is stored in a user-specified file in this implementation,
+/// and the SegmentRef holds the filename.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct FileSegmentRef {
+    path: PathBuf,
+}
+
+#[typetag::serde]
+impl SegmentRef for FileSegmentRef {
+    fn resolve(&self) -> anyhow::Result<Segment> {
+        let mut contents = Vec::new();
+        let mut file = File::open(&self.path)?;
+        file.read_to_end(&mut contents)?;
+        let segment: Segment = bincode::deserialize(&contents)?;
+        Ok(segment)
+    }
+}
+
+impl FileSegmentRef {
+    /// Construct a [FileSegmentRef]
+    ///
+    /// This builds a FileSegmentRef that stores `segment` in a file at `path`.
```

Comment:
This is a little bit of a odd suggestion but should we consider linking to the EVM code in the docstr kinda like: "if you wanna see an example of using this check here ...." ?

Reply:
Interesting! Perhaps a bit odd, but probably a discoverability win so I like it :+1: